pkg/provisioner/ironic/ironic.go

```diff
@@ -477,13 +477,16 @@ func getNICDetails(ifdata []introspection.InterfaceType,
 	for i, intf := range ifdata {
 		baseIntf := basedata[intf.Name]
 		vlans, vlanid := getVLANs(baseIntf)
-
+		ip := intf.IPV4Address
+		if ip == "" {
+			ip = intf.IPV6Address
+		}
```

Comment:
This seems fine, but it actually highlights another bug that will remain.

Once we try this with a dual-stack IPv4+IPv6 environment, we will now be missing the IPv4 address for an interface.

Maybe we should actually change the API here a bit to be able to list more than one IP address?

Reply:
I'm also OK with filing a bug to track fixing dual-stack / multiple IPs in this API.

The PR as-is is an improvement

Reply:
Yeah I did wonder about dual stack but as you say it's probably an API change unless we append a duplicate nic with the same name but a different IP?

I'll raise a bug so we can track resolving that, this should at least fix the single stack case so we can make progress.

Reply:
https://github.com/metal3-io/baremetal-operator/issues/458